### PR TITLE
add color logging capabilities for promlog

### PR DESCRIPTION
**Pull request by cstyan**
Add's color logging capabilities to promlog when we're logging to the terminal, otherwise we just get a standard go-kit logger when logging to a file. 
We'll have to update deps/vendor packages in prometheus after we merge.

cc: @krasi-georgiev @juliusv

### CODE REVIEW

**Comment by cstyan** [comment]
cc: @fabxc

**Comment by krasi-georgiev** [comment]
Did you check if colours can be applied to selected name=value groups rather than the entire line?

**Comment by cstyan** [comment]
AFAICT, that would require a change to the `go-kit/log/term` package. So unless anyone thinks it's really necessary, I'll just leave the colors change as is for now.

**Comment by brian-brazil** [comment]
Is this still wanted?

**Comment by SuperQ** [comment]
This PR seems fine, but it needs a rebase for the change to `go-kit/log`

File: promlog/log.go [hunk 1]
```diff
@@ -31,6 +32,25 @@ type AllowedLevel struct {
...
+			continue
+		}
+		switch keyvals[i+1] {
+		case "debug":
+			return term.FgBgColor{Fg: term.Blue}
+		case "warn":
+			return term.FgBgColor{Fg: term.Yellow}
+		case "error":
+			return term.FgBgColor{Fg: term.Red}
+		default:
```
Comment (juliusv) [inline promlog/log.go:47]: How about green for INFO?

Comment (SuperQ) [reply promlog/log.go:47]: Sure

**Comment by SuperQ** [comment]
This package is deprecated (https://github.com/prometheus/common/pull/720).

**Pull request closed without merging**